File: boj/dp/9095.cpp
#include <iostream>

#define endl '\n'
#define fastio cin.sync_with_stdio(false); cin.tie(nullptr);

using namespace std;

int d[11];
int dp(int n){
    if(n==1) return 1;
    if(n==2) return 2;
    if(n==3) return 4;
    if(d[n]!=0) return d[n];
    return d[n] = dp(n-1) + dp(n-2) + dp(n-3);
}

int main(){
    fastio;
    
    int test_case,n;
    cin >> test_case;
    while(test_case--){
        cin >> n;
        cout << dp(n) << endl;
    }
    
    return 0;
}

File: boj/dp/9095.py
test_case = int(input())

d = []
d.append(0)
d.append(1)
d.append(2)
d.append(4)

for i in range(4,11):
    d.append(d[i-1] + d[i-2] + d[i-3])

for _ in range(test_case):
    n = int(input())
    print(d[n])

